common/compositor/gl/gl_shader_pre_built/bin_to_c_array.c

```diff
@@ -0,0 +1,85 @@
+#include<stdio.h>
+#include<stdlib.h>
+
+#define LICENSE_TERM \
+"// Copyright (c) 2017 Intel Corporation\n"\
+"//\n"\
+"// Licensed under the Apache License, Version 2.0 (the \"License\");\n"\
+"// you may not use this file except in compliance with the License.\n"\
+"// You may obtain a copy of the License at\n"\
+"//\n"\
+"//      http://www.apache.org/licenses/LICENSE-2.0\n"\
+"//\n"\
+"// Unless required by applicable law or agreed to in writing, software\n"\
+"// distributed under the License is distributed on an \"AS IS\" BASIS,\n"\
+"// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n"\
+"// See the License for the specific language governing permissions and\n"\
+"// limitations under the License.\n"\
+"\n"\
+"// clang-format off\n"
+
+void print_usage(void)
+{
+  fprintf(stdout, "./bin_to_c_array <input bin file> <output file>\n");
+  fprintf(stdout, "ex)./bin_to_c_array hwc_shader_prog_1.shader_test.bin hwc_shader_prog_1.h\n\n");
+}
+
+int main(int argc, char *argv[])
+{
+  FILE *binary_fp, *header_fp;
+  int byte;
+  int i = 0;
+
+  if (argc < 3) {
+    fprintf(stderr, "need input and output file names\n");
+    print_usage();
+    exit(-1);
+  }
+
+  /* print out information */
+  fprintf(stdout, "input file:%s\n", argv[1]);
+  fprintf(stdout, "output file:%s\n", argv[2]);
+
+  binary_fp = fopen(argv[1], "rb");
+  header_fp = fopen(argv[2], "wb");
+
+  if (!binary_fp) {
+    fprintf(stderr, "input file does not exist\n");
+    print_usage();
+    exit(-1);
```

Comment:
fclose any opened handles before exiting.

Reply:
Please review this new file. It is now closing all file ptrs.

Reply:
Reviewed and file handles are properly managed now.